File: Finance.Domain/Entities/Transaction.cs
using Finance.Domain.Common;

namespace Finance.Domain.Entities;

public class Transaction : BaseEntity
{
    public decimal Amount { get; set; }
    public string Currency { get; set; } = null!;
    public string? Note { get; set; }
    public DateTime Date { get; set; } = DateTime.UtcNow;
    public TransactionType Type { get; set; }
    public decimal ExchangeRate { get; set; }
    
    public int AccountId { get; set; }
    public Account? Account { get; set; }
    
    public int? CategoryId { get; set; }
    public Category? Category { get; set; }
    
    public int? CustomCategoryId { get; set; }
    public CustomCategory? CustomCategory { get; set; }
}

File: frontend/src/app/modules/dashboard/resources/models/transaction.ts
import { Account } from "../../../transactions/resources/models/account-model"
import { Category } from "../../../transactions/resources/models/category"

export interface Transaction{
  id: number,
  note: string | null,
  date: Date,
  amount: number,
  type: TransactionType,
  currency: string,
  category: Category
  account: Account
}

export enum TransactionType{
  Income,
  Expense
}
